packages/notifi-dataplane/lib/types/FusionMessage.ts

```diff
@@ -24,3 +24,20 @@ export interface FusionMessage {
     }>
   >;
 }
+
+export type OptionalTargetType = 'Sms' | 'Telegram' | 'Discord' | 'Email';
+
+export type VariablesJsonPayload = { Platform: TargetVariables } & {
+  [key in OptionalTargetType]?: TargetVariables;
+};
+
+/**
+ * @param message - only plain text is supported.
+ * @param message__markdown - markdown string here will be converted to html.
+ */
+export type TargetVariables = {
+  subject?: string;
+  message?: string;
+  message__markdown?: string;
+  [key: string]: string | undefined;
+};
```

Comment:
This type shape is required for Community Manager posts, but not for OffChain API events. Offchain API events can be any JSON as long as the template is prepared for it.

Reply:
@hans-notifi Right ... So we make all field optional in `TargetVariables`. And also allow to put in whatever key-value in it. Or is there any other concern?

Reply:
As discussed in Slack, the platform discriminators are not needed for OffChain API events either. You can literally put in any valid JSON and it will work as long as you have a matching template - and for some message types, we would want to do exactly that in order to avoid duplicating data for each platform we want to send on.